test/lightning/jobs_test.exs

```diff
@@ -101,6 +104,146 @@ defmodule Lightning.JobsTest do
+               count_workflows_after
+
+      assert Enum.member?(
+               Enum.map(workflows_before, fn w -> w.id end),
+               job_1.workflow_id
+             )
+             |> Kernel.not()
+
+      assert Enum.member?(
+               Enum.map(workflows_before, fn w -> w.id end),
+               job_2.workflow_id
+             )
+             |> Kernel.not()
+
+      assert Enum.member?(
+               Enum.map(workflows_after, fn w -> w.id end),
+               job_1.workflow_id
+             )
+
+      assert Enum.member?(
+               Enum.map(workflows_after, fn w -> w.id end),
+               job_2.workflow_id
+             )
+    end
+
+    test "create_job/1 with an upstream job returns a job with the upstream job's workflow_id" do
+      project_id = project_fixture().id
+
+      {:ok, %Job{} = parent_job} =
+        Jobs.create_job(%{
+          body: "some body",
+          enabled: true,
+          name: "some name",
+          adaptor: "@openfn/language-common",
+          trigger: %{type: "webhook"},
+          project_id: project_id
+        })
+
+      {:ok, %Job{} = job} =
+        Jobs.create_job(%{
+          body: "some body",
+          enabled: true,
+          name: "some name",
+          adaptor: "@openfn/language-common",
+          trigger: %{type: "on_job_success", upstream_job_id: parent_job.id},
+          project_id: project_id
+        })
+
+      assert job.workflow_id == parent_job.workflow_id
+    end
+
+    test "update_job/2 from a cron to a webhook trigger does NOT create a new workflow" do
+      project_id = project_fixture().id
+
+      {:ok, %Job{} = job} =
+        Jobs.create_job(%{
+          body: "some body",
+          enabled: true,
+          name: "some name",
+          adaptor: "@openfn/language-common",
+          trigger: %{type: "cron", cron_expression: "* * * *"},
+          project_id: project_id
+        })
+
+      count_workflows_before = Workflows.list_workflows() |> Enum.count()
+
+      Jobs.update_job(job, %{trigger: %{id: job.trigger.id, type: "webhook"}})
+
+      assert count_workflows_before == Workflows.list_workflows() |> Enum.count()
+    end
+
+    test "update_job/2 from upstream_job A (in workflow 1) to upstream_job B (in workflow 2) changes the updated job's workflow_id to 2" do
+      project_id = project_fixture().id
+
+      {:ok, %Job{} = parent_job_1} =
+        Jobs.create_job(%{
+          body: "some body",
+          enabled: true,
+          name: "some name",
+          adaptor: "@openfn/language-common",
+          trigger: %{type: "cron"},
+          project_id: project_id
+        })
+
+      {:ok, %Job{} = parent_job_2} =
+        Jobs.create_job(%{
+          body: "some body",
+          enabled: true,
+          name: "some name",
+          adaptor: "@openfn/language-common",
+          trigger: %{type: "cron"},
+          project_id: project_id
+        })
+
+      {:ok, %Job{} = upstream_job_a} =
```

Comment:
this is a "downstream" job, not upstream, right?

Reply:
Yes it is. will change it.

Reply:
Done